components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java

```diff
@@ -3568,6 +3582,40 @@ public void setGlobalRbacScopeIssuerEnabled(boolean globalRbacScopeIssuerEnabled
         this.globalRbacScopeIssuerEnabled = globalRbacScopeIssuerEnabled;
     }
 
+    public List<String> getSupportedTokenEndpointSigningAlgorithms() {
+
+        return supportedTokenEndpointSigningAlgorithms;
+    }
+
+    /**
+     * Parse supported signing algorithms and add them to the supportedTokenEndpointSigningAlgorithms list.
+     *
+     * @param algorithms OMElement of supported algorithms.
+     */
+    private void parseSupportedTokenEndpointSigningAlgorithms(OMElement algorithms)
+            throws ServerConfigurationException {
+
+        if (algorithms == null) {
+            return;
+        }
+
+        Iterator iterator = algorithms.getChildrenWithLocalName(
+                ConfigElements.SUPPORTED_TOKEN_ENDPOINT_SIGNING_ALG);
+        if (iterator != null) {
+            for (; iterator.hasNext(); ) {
```

Comment:
lets replace with a while loop. improves readability

```suggestion
            while (iterator.hasNext()) {
```